edx_notifications/server/web/static/edx_notifications/js/views/counter_icon_view.js

```diff
@@ -0,0 +1,66 @@
+var CounterIconView = Backbone.View.extend({
+    initialize: function(options){
+        this.options = options;
+
+        /* re-render if the model changes */
+        this.listenTo(this.model,'change', this.modelChanged);
+
+        /* make the async call to the backend REST API */
+        /* after it loads, the listenTo event will file and */
+        /* will call into the rendering */
+        this.model.fetch();
+    },
+
+    events: {
+        'click .edx-notifications-icon': 'showPane'
+    },
+
+    /* cached notification_icon.html template */
+    fetched_template: null,
+
+    /* cached notifications pane view */
+    notification_pane: null,
+
+    modelChanged: function() {
+        this.render();
+    },
+
+    render: function () {
+        if (!this.fetched_template) {
+            var self = this;
+
+            /* load Underscore template asynchronously on first load */
+            $.get("/static/edx_notifications/templates/underscore/notification_icon.html")
```

Comment:
Our best practice (that we haven't managed to start using yet!) is to use RequireJS.text to load Underscore templates:

https://github.com/requirejs/text

The beauty of this is that it works with RequireJS Optimizer so that the templates themselves get included in the single JS file that is minified for the entire page.